src/smart-components/order/orders-list.js

```diff
@@ -1,78 +1,66 @@
+const OrdersList = () => {
   const [ isFetching, setFetching ] = useState(true);
   const [ searchValue, setSearchValue ] = useState('');
-  const { data, meta } = useSelector(({ orderReducer }) => orderReducer[type]);
+  const { data, meta } = useSelector(({ orderReducer }) => orderReducer.orders);
   const portfolioItems = useSelector(({ portfolioReducer: { portfolioItems }}) => portfolioItems.data);
   const dispatch = useDispatch();
   useEffect(() => {
     setFetching(true);
-    Promise.all([ dispatch(apiRequest[type]()), dispatch(fetchPlatforms()) ])
+    Promise.all([ dispatch(fetchOrders()), dispatch(fetchPlatforms()) ])
     .then(() => setFetching(false));
-  }, [ type ]);
+  }, []);
 
-  const handlePagination = (...args) => {
+  const handlePagination = (_apiProps, pagination) => {
     setFetching(true);
-    dispatch(apiRequest[type](...args))
+    dispatch(fetchOrders(pagination))
     .then(() => setFetching(false))
     .catch(() => setFetching(false));
   };
 
   return (
-    <Fragment>
-      <Route path={ `${url}/:orderItemId/messages` } render={ (props) => <OrderMessagesModal closeUrl={ url } { ...props } /> } />
-      <div className="pf-u-pb-md pf-u-pl-xl pf-u-pr-xl orders-list">
-        <Level>
-          <LevelItem className="pf-u-mt-md">
-            <FilterToolbarItem searchValue={ searchValue } onFilterChange={ value => setSearchValue(value) } placeholder="Filter by name..." />
-          </LevelItem>
-          <LevelItem>
-            <AsyncPagination isDisabled={ isFetching } apiRequest={ handlePagination } meta={ meta } />
-          </LevelItem>
-        </Level>
-      </div>
-      <DataList aria-label={ type }>
-        { isFetching
-          ? <ListLoader />
-          : data
-          .filter(item => getOrderPortfolioName(item, portfolioItems).toLowerCase().includes(searchValue.toLowerCase()))
-          .map((item, index) => (
-            <OrderItem
-              key={ item.id }
-              index={ index }
-              type={ type }
-              item={ item }
-            />
-          )) }
-      </DataList>
-    </Fragment>
+    <Grid gutter="md">
+      <GridItem>
+        <Section type="content">
+          <div className="pf-u-pb-md pf-u-pl-xl pf-u-pr-xl orders-list">
+            <Level>
+              <LevelItem className="pf-u-mt-md">
+                <FilterToolbarItem searchValue={ searchValue } onFilterChange={ value => setSearchValue(value) } placeholder="Filter by name..." />
+              </LevelItem>
+              <LevelItem>
+                <AsyncPagination isDisabled={ isFetching } apiRequest={ handlePagination } meta={ meta } />
+              </LevelItem>
+            </Level>
+          </div>
+          <DataList aria-label="order-list">
+            { isFetching
+              ? <ListLoader />
+              : data
+              .filter(item => getOrderPortfolioName(item, portfolioItems).toLowerCase().includes(searchValue.toLowerCase()))
```

Comment:
I would extract this into a function and put it above, it is hard to read

Reply:
This will be removed. Filtering will be done via API once this is merged.